assets/app/scripts/controllers/javaLink.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .run(function(ProxyPod, BaseLink, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="container.jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView(container, container.jolokiaUrl)" href="">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+      var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
```

Comment:
looking for a parent scope with `pod` set is actually pretty weird here... this is the podTemplate, which can have 0..* actual pods associated with it.